tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,209 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run open-vm-tools testing against VMware ESXi
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_autotest::esxi_utils;
+use Time::Local;
+
+our $VM_POWERED_ON  = 'Powered on';
+our $VM_POWERED_OFF = 'Powered off';
+
+sub run {
+    assert_script_run('rpmquery open-vm-tools');
+    systemctl('status vmtoolsd');
+    systemctl('status vgauthd');
+
+    my $vm_name = console('svirt')->name;
+    my $vm_id   = esxi_vm_get_vmid($vm_name);
+    my $vm_ip   = esxi_vm_public_ip($vm_id);
+
+    record_info('Guest Name', $vm_name);
+    record_info('Guest ID',   $vm_id);
+    record_info('Guest IP',   $vm_ip);
+
+    die "The variable \$vm_id or \$vm_ip cannot be empty." if ($vm_id eq "" || $vm_ip eq "");
+
+    do_power_management_tests($vm_id, $vm_ip);
+    do_networking_tests($vm_name, $vm_id, $vm_ip);
+    do_clock_sync_tests($vm_name, $vm_id, $vm_ip);
+}
+
+sub check_vm_power_state {
+    my ($vm_id, $vm_ip, $power_action, $if_pingable, $check_vm_state, $if_vm_state, $expect_vm_state) = @_;
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/$if_vm_state/) {
+        esxi_vm_power_cycle($vm_id, $power_action);
+        wait_for_vm_bootup($vm_ip, $if_pingable);
+    } elsif ($check_vm_state) {
```

Comment:
I am not very sure what this "elsif ($check_vm_state)" is used for ? It seems that it can be removed ?